internal/gengapic/gengapic.go

```diff
@@ -281,98 +281,102 @@ func (g *generator) getFormattedValue(m *descriptor.MethodDescriptorProto, field
 	return value, nil
 }
 
-func (g *generator) insertImplicitRequestHeaders(m *descriptor.MethodDescriptorProto) error {
-	headers, err := parseImplicitRequestHeaders(m)
-	if err != nil {
-		return err
+func (g *generator) insertImplicitRequestHeaders(m *descriptor.MethodDescriptorProto, headers [][]string) error {
+	if len(headers) == 0 {
+		return nil
 	}
-	if len(headers) > 0 {
-		seen := map[string]bool{}
-		var formats, values strings.Builder
-		for _, h := range headers {
-			field := h[1]
-			// skip fields that have multiple patterns, they use the same accessor
-			if _, dupe := seen[field]; dupe {
-				continue
-			}
-			seen[field] = true
-			accessor := fmt.Sprintf("req%s", fieldGetter(field))
-			accessor, err = g.getFormattedValue(m, field, accessor)
-			if err != nil {
-				return err
-			}
-			fmt.Fprintf(&values, " %q, %s,", url.QueryEscape(field), accessor)
-			formats.WriteString("%s=%v&")
+
+	seen := map[string]bool{}
+	var formats, values strings.Builder
+	for _, h := range headers {
+		field := h[1]
+		// skip fields that have multiple patterns, they use the same accessor
+		if _, dupe := seen[field]; dupe {
+			continue
 		}
-		// Trim the trailing comma and ampersand symbols.
-		f := formats.String()[:formats.Len()-1]
-		v := values.String()[:values.Len()-1]
-		g.printf("md := metadata.Pairs(\"x-goog-request-params\", fmt.Sprintf(%q,%s))", f, v)
+		seen[field] = true
+		accessor := fmt.Sprintf("req%s", fieldGetter(field))
+		accessor, err := g.getFormattedValue(m, field, accessor)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(&values, " %q, %s,", url.QueryEscape(field), accessor)
+		formats.WriteString("%s=%v&")
 	}
+	// Trim the trailing comma and ampersand symbols.
+	f := formats.String()[:formats.Len()-1]
+	v := values.String()[:values.Len()-1]
+	g.printf("md := metadata.Pairs(\"x-goog-request-params\", fmt.Sprintf(%q,%s))", f, v)
 	return nil
 }
 
-func (g *generator) insertDynamicRequestHeaders(m *descriptor.MethodDescriptorProto) error {
-	headers, err := parseDynamicRequestHeaders(m)
-	if err != nil {
-		return err
+func (g *generator) insertDynamicRequestHeaders(m *descriptor.MethodDescriptorProto, headers [][]string) error {
+	if len(headers) == 0 {
+		return nil
 	}
-	if len(headers) > 0 {
```

Comment:
I didn't change this code, just de-dented it by checking the opposite case and returning early.